pkg/libvirttools/libvirt_domain.go

```diff
@@ -276,11 +287,62 @@ func (domain *libvirtDomain) SetVcpus(vcpus uint) error {
 	return domain.d.SetVcpusFlags(vcpus, libvirt.DOMAIN_VCPU_CONFIG|libvirt.DOMAIN_VCPU_LIVE)
 }
 
+// TODO: move this to a helper function file
+func determineNumberOfDeviceNeeded (memChangeInKib int64, isAttach bool) int {
+	var numberMemoryDevicesNeeded int
+
+	temp := math.Abs(float64(memChangeInKib) ) / float64(memoryDeviceSizeInKiB)
+	if temp < 1 {
+		if isAttach {
+			numberMemoryDevicesNeeded = 1
+		} else {
+			numberMemoryDevicesNeeded = 0
+		}
+	} else {
+		if isAttach{
+			numberMemoryDevicesNeeded = int(math.Ceil(temp))
+		} else {
+			numberMemoryDevicesNeeded = int(math.Floor(temp))
+		}
+	}
+
+	return numberMemoryDevicesNeeded
+}
+
 // Update domain current memory
-func (domain *libvirtDomain) SetCurrentMemory(memInKib uint64) error {
-	return domain.d.SetMemoryFlags(memInKib, libvirt.DOMAIN_MEM_CONFIG|libvirt.DOMAIN_MEM_LIVE)
+// the memory device is 128 Mib each
+func (domain *libvirtDomain) SetCurrentMemory(memChangeInKib int64) error {
+   glog.V(4).Infof("MemoryChanges in KiB: %v", memChangeInKib)
+	// no memory changes, just return
+	if memChangeInKib == 0 {
```

Comment:
Actually no memory will be detached if memChangeInKib is smaller than a certain value (currently 128M). Should we also check this and return early with an error here?

Reply:
will add. but this is bigger than just treating this as an error case.
if the hutplug/unplug approach is eventually the ONLY way to support vertical scaling. the minimal device size will need to be aware to VPA so it will round the request to match it. or the round-up is done at the runtime.